packages/client/hooks/useLinearProjectsAndTeams.ts

```diff
@@ -0,0 +1,154 @@
+import graphql from 'babel-plugin-relay/macro'
+import {useMemo, useState} from 'react'
+import {useFragment} from 'react-relay'
+import {
+  useLinearProjectsAndTeams_teamMember$data,
+  useLinearProjectsAndTeams_teamMember$key
+} from '../__generated__/useLinearProjectsAndTeams_teamMember.graphql'
+import getNonNullEdges from '../utils/getNonNullEdges'
+import getUniqueEdges from '../utils/getUniqueEdges'
+import {isNotNull} from '../utils/predicates'
+
+type ArrayElement<ArrayType extends readonly unknown[] | null | undefined> =
+  ArrayType extends readonly (infer ElementType)[] ? ElementType : never
+
+type LinearQuery = useLinearProjectsAndTeams_teamMember$data extends {
+  integrations?: {linear?: {api?: {query?: infer Q | null} | null} | null} | null
+}
+  ? Q
+  : never
+
+type AllProjectEdges = LinearQuery extends {allProjects?: {edges?: infer E | null} | null}
+  ? E
+  : never
+
+type TeamEdges = LinearQuery extends {teams?: {edges?: infer E | null} | null} ? E : never
+
+type ProjectEdge = NonNullable<ArrayElement<AllProjectEdges>>
+type TeamEdge = NonNullable<ArrayElement<TeamEdges>>
+
+export type Project = NonNullable<ProjectEdge['node']>
+export type Team = NonNullable<TeamEdge['node']>
+
+export type LinearProjectOrTeam = Project | Team
```

Comment:
+1 We probably don't need all these types? Also these appear to be redefined in multiple other places.